fail2ban/client/fail2banclient.py

```diff
@@ -0,0 +1,462 @@
+						 "remove " + self._conf["socket"] + ". If "
+						 "you used fail2ban-client to start the "
+						 "server, adding the -x option will do it")
+			if self._server:
+				self._server.quit()
+			return False
+		return True
+
+	def __waitOnServer(self, alive=True, maxtime=None):
+		if maxtime is None:
+			maxtime = self._conf["timeout"]
+		# Wait for the server to start (the server has 30 seconds to answer ping)
+		starttime = time.time()
+		logSys.debug("__waitOnServer: %r", (alive, maxtime))
+		test = lambda: os.path.exists(self._conf["socket"]) and self.__ping() 
+		with VisualWait(self._conf["verbose"]) as vis:
+			sltime = 0.0125 / 2
+			while self._alive:
+				runf = test()
+				if runf == alive:
+					return True
+				now = time.time()
+				# Wonderful visual :)
+				if now > starttime + 1:
+					vis.heartbeat()
+				# f end time reached:
+				if now - starttime >= maxtime:
+					raise ServerExecutionException("Failed to start server")
+				sltime = min(sltime * 2, 0.5)
+				time.sleep(sltime)
+		return False
+
+	def start(self, argv):
+		# Install signal handlers
+		_prev_signals = {}
+		if _thread_name() == '_MainThread':
+			for s in (signal.SIGTERM, signal.SIGINT):
+				_prev_signals[s] = signal.getsignal(s)
+				signal.signal(s, self.__sigTERMhandler)
+		try:
+			# Command line options
+			if self._argv is None:
+				ret = self.initCmdLine(argv)
+				if ret is not None:
+					if ret:
+						return True
+					raise ServerExecutionException("Init of command line failed")
+
+			# Commands
+			args = self._args
+
+			# Interactive mode
+			if self._conf.get("interactive", False):
+				# no readline in test:
+				if PRODUCTION: # pragma: no cover
+					try:
+						import readline
+					except ImportError:
+						raise ServerExecutionException("Readline not available")
+				try:
+					ret = True
+					if len(args) > 0:
+						ret = self.__processCommand(args)
+					if ret:
+						if PRODUCTION: # pragma: no cover
+							readline.parse_and_bind("tab: complete")
+						self.dispInteractive()
+						while True:
+							cmd = input_command()
+							if cmd == "exit" or cmd == "quit":
+								# Exit
+								return True
+							if cmd == "help":
+								self.dispUsage()
+							elif not cmd == "":
+								try:
+									self.__processCommand(shlex.split(cmd))
+								except Exception, e: # pragma: no cover
+									if self._conf["verbose"] > 1:
+										logSys.exception(e)
+									else:
+										logSys.error(e)
+				except (EOFError, KeyboardInterrupt): # pragma: no cover
+					output("")
+					raise
+			# Single command mode
+			else:
+				if len(args) < 1:
+					self.dispUsage()
+					return False
+				return self.__processCommand(args)
+		except Exception as e:
+			if self._conf["verbose"] > 1:
+				logSys.exception(e)
+			else:
+				logSys.error(e)
+			return False
+		finally:
+			self._alive = False
+			for s, sh in _prev_signals.iteritems():
+				signal.signal(s, sh)
+
+
+##
+# Wonderful visual :)
+#
+
+class _VisualWait:
```

Comment:
Some docs would be really appreciated, otherwise I am just guessing what is the intended use for these two